### ISSUE TITLE
`azurerm_app_configuration_feature` no longer allows colon in key which is valid - (uses key from name)

### ISSUE THREAD

**Issue by ElvenSpellmaker** [hashicorp/terraform-provider-azurerm#20985]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

N/A

### AzureRM Provider Version

3.30+

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration_feature

### Terraform Configuration Files

```hcl
# Contrived small example without calculation of the data structure...
locals {
  flattened_feature_toggles = ["foo:bar:baz"]
}

resource "azurerm_app_configuration_feature" "features" {
  for_each = local.flattened_feature_toggles

  configuration_store_id = data.azurerm_app_configuration.default.id
  name                   = each.key
  description            = each.value.description
  label                  = each.key

  tags = var.resource_tags

  lifecycle {
    ignore_changes = [
      enabled,
    ]
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: character ":" is not allowed in "name"
│
│   with azurerm_app_configuration_feature.features["foo:bar:baz"],
│   on app_configuration.tf line 38, in resource "azurerm_app_configuration_feature" "
features":
│   38:   name                   = each.key
│
╵
```

### Expected Behaviour

To be able to specify a key, or sanitise only the name field automatically.

The _key_ field is allowed to contain colons and indeed Microsoft suggest this themselves: https://learn.microsoft.com/en-us/azure/azure-app-configuration/manage-feature-flags#create-feature-flags
![image](https://user-images.githubusercontent.com/2286713/225630443-5e0967f2-de17-420e-b423-f005beaba773.png)

However by "fixing" the name to no longer allow colons (even though the API still accepts them, it's just the portal that doesn't) it's now broken anyone using colons pre 3.30.0 and with no `key` property on the resource we can't migrate to a newer provider without breaking exisitng applications and having to make changes to every single other application.

tl;dr: We need an optional `key` property on the resource which _can_ take colons.

### Actual Behaviour

```
Error: character ":" is not allowed in "name"
```

Due to this issue being solved: https://github.com/hashicorp/terraform-provider-azurerm/issues/19027

### Steps to Reproduce

`terraform apply`

### Important Factoids

No

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/19027
https://github.com/hashicorp/terraform-provider-azurerm/pull/19063
https://learn.microsoft.com/en-us/azure/azure-app-configuration/manage-feature-flags#create-feature-flags

**Comment by teowa** [comment]
Hi @ElvenSpellmaker, thanks for submitting this!
Current provider use the `name` as the value of the `key` compulsively, but `:` character is not supported in `name` from Azure Portal, we should export the `key` property for the `azurerm_app_configuration_feature` which can be customized.

**Comment by ElvenSpellmaker** [comment]
This would definitely help as I can't upgrade beyond 3.29 due to this breaking change.
